docs/configuration.md

```diff
@@ -1,108 +1,114 @@
-# Run-time configuration
+# RedisGears Configuration
+RedisGears provides configuration options that control its operation. These options can be set when the module is bootstrapped and in some cases also during runtime.
 
-RedisGears supports a few run-time configuration options that should be determined when loading the module. In time more options will be added.
+The following sections describe the configuration options the means for setting them.
 
-## Passing Configuration Options During Loading
+**Bootstrap Configuration**
 
-In general, passing configuration options is done by appending arguments after the `--loadmodule` argument in the command line, `loadmodule` configuration directive in a Redis config file, or the `MODULE LOAD` command. For example:
+Configuration options can be set when the module is loaded. The options are passed as a list of option names and their respective values. Configuration is supported both when using the `loadmodule` configuration directive as well as via the [Redis `MODULE LOAD` command](https://redis.io/commands/module-load).
 
-In redis.conf:
+!!! example "Example: Setting configuration options"
+    For setting the module's configuration options from the command line use:
 
-```
-loadmodule redisgears.so OPT1 OPT2
-```
+    ```
+    redis-server --loadmodule /path/to/redisgears.so <option> <value> ...
+    ```
 
-From redis-cli:
+    For setting the module's configuration options in with .conf file use the following format:
 
-```
-127.0.0.6379> MODULE load redisgears.so OPT1 OPT2
-```
+    ```
+    loadmodule /path/to/redisgears.so <option> <value> ...
+    ```
 
-From command line:
+    For setting the module's configuration with the [`MODULE LOAD`](https://redis.io/commands/module-load) command use:
 
-```
-$ redis-server --loadmodule ./redisgears.so OPT1 OPT2
-```
+    ```
+    127.0.0.1:6379> loadmodule /path/to/redisgears.so <option> <value> ...
+    ```
 
-## Passing configuration options at runtime
+**Runtime Configuration**
 
-It is possible to modify certain configuration parameters at runtime using `RG.CONFIGSET` command. The command receives the configuration parameter's name and its value. For example, the following will enable the excution profiler:
-```
-$ RG.CONFIGSET ProfileExecutions 1
-```
+Some configuration options may be set at runtime. Refer to each option's description for runtime configurability.
+
+!!! abstract "Related commands"
+    The following RedisGears commands are related to configuration:
 
-## RedisGears configuration options
+    * [`RG.CONFIGGET`](commands.md#rgconfigget)
+    * [`RG.CONFIGSET`](commands.md#rgconfigset)
 
 ## MaxExecutions
+The **MaxExecutions** configuration option controls the maximal number of executions that will be saved in the executions list. Once this threshold value is reached, older executions will be deleted from the list by order of their creation (FIFO). Only executions with a 'done' status (?) are deleted.
```

Comment:
Regarding (?) yes its true we do not drop executions which are not done even if we exceeded the MaxExecution threshold

Reply:
What about 'aborted'?

Reply:
Rights any execution that finished running (either done or abort will be deleted)